File: app/src/main/java/com/eason/BaseActivity.java
package com.eason;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends base.BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        init();
    }

    protected abstract void init();

    public abstract int getLayoutId();
}

File: app/src/main/java/com/eason/BaseActivity.kt
package com.eason

import android.os.Bundle

abstract class BaseActivity : base.BaseActivity() {

    abstract val layoutId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
        init()
    }

    protected abstract fun init()
}
